pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -973,20 +979,29 @@ func GetBackupFilename(backup *v3.EtcdBackup) string {
 	return snapshot
 }
 
-// transform an imported cluster into a k3s cluster using its discovered version
-func k3sClusterConfig(cluster *v3.Cluster) error {
+// transform an imported cluster into a k3s or k3os cluster using its discovered version
+func k3sClusterConfig(cluster *v3.Cluster, nodes []*v3.Node) error {
 	// version is not found until cluster is provisioned
-	if cluster.Status.Driver == "" || cluster.Status.Version == nil {
+	if cluster.Status.Driver == "" || cluster.Status.Version == nil || len(nodes) == 0 {
 		return &controller.ForgetError{
 			Err:    fmt.Errorf("waiting for full cluster configuration"),
 			Reason: "Pending"}
 	}
-	if cluster.Status.Driver == v3.ClusterDriverK3s {
+	if cluster.Status.Driver == v3.ClusterDriverK3s ||
+		cluster.Status.Driver == v3.ClusterDriverK3os {
 		return nil //no-op
 	}
 
 	if strings.Contains(cluster.Status.Version.String(), "k3s") {
-		cluster.Status.Driver = v3.ClusterDriverK3s
+		for _, node := range nodes {
+			if _, ok := node.Status.NodeLabels["k3os.io/mode"]; ok {
```

Comment:
Can we confirm with @dweomer that this is the best way to determine if this is a k3os cluster? Also, can we end up in some hybrid setup with k3s management running some k3os downstream workers?

Reply:
So, the presence of the `k3os.io/mode` label on a node is the most reliable way to know if it is a k3OS node but, as per the second part of @dramich's comment, does one k3OS node make the entire (possibly hetergeneous) cluster a k3OS cluster? It might make sense to include in the `k3s-[master|worker]` plans to exclude anything with the `k3os.io/mode` label

Reply:
Based on an offline (online) conversation with @daxmc99 we can't handle k3os yet, so if one node is k3os that would stop the upgrade from completing thus getting us back into the same place. Safe bet until this can be fully integrated is call the whole cluster a k3os cluster.